### PR TITLE
Store API: allow searching by `slug` and include `slug` in the response

### LINKED ISSUE #8985: Store API - Add product slug field
## Is your feature request related to a problem? Please describe.
I would like to use woocommerce store api to build a static headless shop without API authentification (using the "Store API" instead of the Woo Rest API).
For that i need the product slug to be available and usable in order to correctly build my urls (routes).

## Describe the solution you'd like
- `/wc/store/v1/products` should return `slug` attribute (in addition of `permalink`)
- we should be able to get a product from his slug and not only by the id (`/wc/store/v1/products/:id` and `/wc/store/v1/products/:slug`). Or to find a product using the search api : `/wc/store/v1/products?slug=product-slug`

### PR DESCRIPTION
**Pull request by albarin**
This PR adds multiple improvements to the Store API:
- New endpoint to return a product by slug `/wc/store/v1/products/:slug`
- Returns a new field `slug` from `/wc/store/v1/products` and `/wc/store/v1/products/:id`
- Allows searching products by slug with `/wc/store/v1/products?slug=first-slug,second-slug`

Fixes https://github.com/woocommerce/woocommerce-blocks/issues/8985

#### Automated Tests
* [ ] Changes in this PR are covered by Automated Tests.
  * [x] Unit tests
  * [ ] E2E tests

#### User Facing Testing

1. Go to `/wc/store/v1/products` and check the returned json includes the product `slug`.
2. Go to `/wc/store/v1/products/:id` (where `:id` is the ID of a product in your site) and check the returned json includes the product `slug`.
3. Go to `/wc/store/v1/products/:slug` (where `:slug` is the slug of a product in your site) and check the product is returned and the returned json includes the product `slug`.

### WooCommerce Visibility

* [x] WooCommerce Core
* [ ] Feature plugin
* [ ] Experimental

### Changelog

> StoreAPI: add new "/wc/store/v1/products/:slug" endpoint, allow searching products by slug with "/wc/store/v1/products?slug=product-slug" and return new field `slug` from the product endpoints.

### CODE REVIEW

File: src/StoreApi/Routes/V1/ProductsBySlug.php [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+	/**
+	 * Get a single item.
+	 *
+	 * @throws RouteException On error.
+	 * @param \WP_REST_Request $request Request object.
+	 * @return \WP_REST_Response
+	 */
+	protected function get_route_response( \WP_REST_Request $request ) {
+		$object = $this->get_product_by_slug( $request['slug'] );
```
Comment (roykho) [inline src/StoreApi/Routes/V1/ProductsBySlug.php]: Perhaps we may want to sanitize the input here? `sanitize_title()` maybe (not tested)?

**Review by roykho — COMMENTED** [review]
I left a small comment inline. Other than that, I wonder if we anything like this for SKUs? If not, that would be a good addition as well perhaps in a separate PR.

Other than that, I've tested the PR and it works as advertise. Good job!

**Comment by albarin** [comment]
> I left a small comment inline. Other than that, I wonder if we anything like this for SKUs? If not, that would be a good addition as well perhaps in a separate PR.
> 
> Other than that, I've tested the PR and it works as advertise. Good job!

Thanks for the review! 🙌 
You can currently search by `sku` with `/wc/store/v1/products?sku=first-sku,second-sku` 🙂

**Review by roykho — APPROVED** [review]
LGTM! Good job!

**Comment by senadir** [comment]
Yay! I had a branch fixing this that I never got it to PR. Thank you for fixing it :D

**Pull request merged**